node/src/api/accounts.rs

```diff
@@ -1,202 +1,167 @@
-    // TODO: get some randomness here to pass it in SecretKey::with_rng method
-    let key_pair = SecretKey::new();
-    let pub_key: PublicKey = key_pair.public_key();
-    let auth_scheme: AuthScheme = AuthScheme::RpoFalcon512 { pub_key };
-
-    // we need to use an initial seed to create the wallet account
-    let init_seed: [u8; 32] = [
-        95, 113, 209, 94, 84, 105, 250, 242, 223, 203, 216, 124, 22, 159, 14, 132, 215, 85, 183, 204,
-        149, 90, 166, 68, 100, 73, 106, 168, 125, 237, 138, 16,
-    ];
-
-    let (game_account, _) = create_basic_aze_player_account(
-        init_seed,
-        auth_scheme,
-        AccountType::RegularAccountImmutableCode
-    ).unwrap();
-
-    Ok(
-        Json(PlayerAccountCreationResponse {
-            is_created: true,
-            account_id: game_account.id().into(),
-        })
-    )
-}
-
-async fn print_account_status(client: &AzeClient, account_id: AccountId) {
-    let (regular_account, _seed) = client.get_account(account_id).unwrap();
-    println!("Account asset count --> {:?}", regular_account.vault().assets().count());
-    println!("Account storage root --> {:?}", regular_account.storage().root());
-    println!("Account slot 100 --> {:?}", regular_account.storage().get_item(100));
-    println!("Account slot 101 --> {:?}", regular_account.storage().get_item(101));
-}
+use aze_lib::accounts::create_basic_aze_player_account;
+use aze_lib::client::{
+    self,
+    create_aze_client,
+    AzeAccountTemplate,
+    AzeClient,
+    AzeGameMethods,
+    AzeTransactionTemplate,
+    SendCardTransactionData,
+};
+use aze_lib::constants::BUY_IN_AMOUNT;
+use aze_lib::notes::{ consume_notes, mint_note };
+use aze_lib::executor::execute_tx_and_sync;
+use aze_lib::utils::GameStorageSlotData;
+
+use aze_types::accounts::{
+    AccountCreationError,
+    AccountCreationResponse,
+    PlayerAccountCreationResponse,
+};
+use aze_lib::utils::log_account_status;
+use miden_lib::AuthScheme;
+use miden_objects::{
+    assets::TokenSymbol,
+    assets::{ Asset, FungibleAsset },
+    crypto::dsa::rpo_falcon512::{ PublicKey, SecretKey },
+    notes::NoteType,
+};
+use miden_client::client::{
+    accounts::{ AccountTemplate, AccountStorageMode },
+    transactions::transaction_request::TransactionTemplate,
+};
+
+use actix_web::{ get, web::Json };
+
+// TODO: pass account id of the players as request object in this game
+#[get("/v1/game/create-account")]
+pub async fn create_aze_game_account() -> Result<
+    Json<AccountCreationResponse>,
+    AccountCreationError
+> {
+    let mut client: AzeClient = create_aze_client();
+    let slot_data = GameStorageSlotData::default();
+
+    // TODO: creating player just for testing purposes
+    let (player_account, _) = client
+        .new_game_account(
+            AzeAccountTemplate::PlayerAccount {
+                mutable_code: false,
+                storage_mode: AccountStorageMode::Local, // for now
+            },
+            slot_data.clone()

```

Comment:
No, If we are creating a player account we should be able to do it as 
```
        client.new_game_account(
            AzeAccountTemplate::PlayerAccount {
                mutable_code: false,
                storage_mode: AccountStorageMode::Local, // for now
            },
            None // we should pass None here instead of slot_data.
       )
            
      ```